Dima/netflix-roulette/src/Pages/Home/Home.tsx

```diff
@@ -30,84 +30,80 @@ const connector = connect(mapStateToProps, mapDispatchToProps)
+    const URLData = parse(props.location.search) as { sortBy: string, search: string, searchBy: string };
     const { sortBy } = URLData
     let requestData = stringify({ ...URLData, search, searchBy })
-    this.props.filmsRequested(0, sortBy, searchBy, search)
-    this.props.history.push({
-      pathname: '/search/',
+    props.filmsRequested({offset: 0, sortBy, searchBy, search})
+    props.history.push({
+      pathname: '/search',
       search: requestData
     });
 
   };
 
-  handleSearchChangeForSort = ({ sortBy }: { sortBy: string }) => {
-    const URLData = parse(this.props.location.search) as { search: string, searchBy: string };
+  const handleSearchChangeForSort = ({ sortBy }: { sortBy: string }) => {
+    const URLData = parse(props.location.search) as { search: string, searchBy: string };
     const { searchBy, search } = URLData
     let requestData = stringify({ ...URLData, sortBy } )
-    console.log(requestData)
-    this.props.filmsRequested(0, sortBy, searchBy, search)
-    this.props.history.push( {
-      pathname: '/search/',
+    props.filmsRequested({offset: 0, sortBy, searchBy, search})
+    props.history.push( {
+      pathname: '/search',
       search: requestData
     });
   };
 
-  handlePagination = (offset: number, pagination: boolean = false) => {
-    const URLData = parse(this.props.location.search) as { 
+  const handlePagination = (offset: number, pagination: boolean = false) => {
+    const URLData = parse(props.location.search) as { 
       sortBy: string, 
       searchBy: string, 
       search: string 
     };
     const { sortBy, searchBy, search } = URLData 
-    this.props.filmsRequested(offset, sortBy, searchBy, search, pagination)
+    console.log(films.length)
+    props.filmsRequested({offset, sortBy, searchBy, search, pagination})
   }
 
-  render() {
-    const { films } = this.props;
+  const { films } = props;
     return (
       <div className="main-wrapp">
         <div className="top">
           <div className="top__wrapp">
             <Header />
             <Search 
-              location = { this.props.location }
-              handleSearchChange = { this.handleSearchChangeForSearch }
+              handleSearchChange = { handleSearchChangeForSearch }
             />
           </div>
         </div>
         <Sort
-          location = { this.props.location }
-          history = {this.props.history}
-          handleSearchChange = { this.handleSearchChangeForSort }
+          handleSearchChange = { handleSearchChangeForSort }
           numberOfFilms = { films.length }
         />
         <InfiniteScroll
-          pageStart={ films.length }
-          loadMore={ () => this.handlePagination(films.length + 10, true) }
-          hasMore={ true }
-          loader={ <div className="loader" key={0}>Loading ...</div> }
+          pageStart = { 1 }
+          loadMore = { () => handlePagination(films.length + 10, true) }
+          hasMore = { true || false }
```

Comment:
WHAT??? You just copied this from an example, as I assume. Hope this wasn't your idea.  Do you realize that this logical expression always returns TRUE.